autoload/lightline/colorscheme/PaperColor.vim

```diff
@@ -0,0 +1,209 @@
+        return 6
+      elseif a:x < 219
+        return 7
+      elseif a:x < 243
+        return 8
+      else
+        return 9
+      endif
+    else
+      if a:x < 14
+        return 0
+      else
+        let l:n = (a:x - 8) / 10
+        let l:m = (a:x - 8) % 10
+        if l:m < 5
+          return l:n
+        else
+          return l:n + 1
+        endif
+      endif
+    endif
+  endfunction
+
+  " Returns the actual grey level represented by the grey index
+  function! s:grey_level(n)
+    if &t_Co == 88
+      if a:n == 0
+        return 0
+      elseif a:n == 1
+        return 46
+      elseif a:n == 2
+        return 92
+      elseif a:n == 3
+        return 115
+      elseif a:n == 4
+        return 139
+      elseif a:n == 5
+        return 162
+      elseif a:n == 6
+        return 185
+      elseif a:n == 7
+        return 208
+      elseif a:n == 8
+        return 231
+      else
+        return 255
+      endif
+    else
+      if a:n == 0
+        return 0
+      else
+        return 8 + (a:n * 10)
+      endif
+    endif
+  endfunction
+
+  " Returns the palette index for the given grey index
+  function! s:grey_color(n)
+    if &t_Co == 88
+      if a:n == 0
+        return 16
+      elseif a:n == 9
+        return 79
+      else
+        return 79 + a:n
+      endif
+    else
+      if a:n == 0
+        return 16
+      elseif a:n == 25
+        return 231
+      else
+        return 231 + a:n
+      endif
+    endif
+  endfunction
+
+  " Returns an approximate color index for the given color level
+  function! s:rgb_number(x)
+    if &t_Co == 88
+      if a:x < 69
+        return 0
+      elseif a:x < 172
+        return 1
+      elseif a:x < 230
+        return 2
+      else
+        return 3
+      endif
+    else
+      if a:x < 75
+        return 0
+      else
+        let l:n = (a:x - 55) / 40
+        let l:m = (a:x - 55) % 40
+        if l:m < 20
+          return l:n
+        else
+          return l:n + 1
+        endif
+      endif
+    endif
+  endfunction
+
+  " Returns the actual color level for the given color index
+  function! s:rgb_level(n)
+    if &t_Co == 88
+      if a:n == 0
+        return 0
+      elseif a:n == 1
+        return 139
+      elseif a:n == 2
+        return 205
+      else
+        return 255
+      endif
+    else
+      if a:n == 0
+        return 0
+      else
+        return 55 + (a:n * 40)
+      endif
+    endif
+  endfunction
+
+  " Returns the palette index for the given R/G/B color indices
+  function! s:rgb_color(x, y, z)
+    if &t_Co == 88
+      return 16 + (a:x * 16) + (a:y * 4) + a:z
+    else
+      return 16 + (a:x * 36) + (a:y * 6) + a:z
+    endif
+  endfunction
+
+  " Returns the palette index to approximate the given R/G/B color levels
+  function! s:color(r, g, b)
+    " Get the closest grey
+    let l:gx = s:grey_number(a:r)
+    let l:gy = s:grey_number(a:g)
+    let l:gz = s:grey_number(a:b)
+
+    " Get the closest color
+    let l:x = s:rgb_number(a:r)
```

Comment:
In this context,  x, y and z don't represent t r, g, b, values exactly but represent just palette index.
These names are used to distinguish palette colors from r, g, b values.
So, I think instead using r,g,b seems to be inappropriate.